torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -575,86 +581,213 @@ void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
+        return loadSub(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::relu"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadRelu(node);
+      }));
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::relu_"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadRelu(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::_convolution"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadConvolution(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::batch_norm"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadBatchNorm(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::max_pool2d"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadMaxPool2d(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::adaptive_avg_pool2d"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadAdaptiveAvgPool2d(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::t"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadTranspose(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::t_"),
+      NodeLoaderInfo({0}, [this](const torch::jit::Node *node) {
+        return loadTranspose(node);
+      }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::exp"),
+      NodeLoaderInfo(
+          {0}, [this](const torch::jit::Node *node) { return loadExp(node); }));
+
+  nodeLoaderMapping_.emplace(
+      at::Symbol::fromQualString("aten::min"),
+      NodeLoaderInfo({0, 1}, [this](const torch::jit::Node *node) {
+        return loadMin(node);
+      }));
+}
 
-  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul")] =
-      [this](const torch::jit::Node *node) { return loadMul(node); };
-  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul_")] =
-      [this](const torch::jit::Node *node) { return loadMul(node); };
+void PyTorchModelLoader::loadNode(const torch::jit::Node *node) {
+  auto kind = node->kind();
+  assert(nodeLoaderMapping_.count(kind));
+  auto &loaderValue = nodeLoaderMapping_.at(kind);
+  ValueMap valueMapReplacements =
+      createConstantPhReplacements(node, loaderValue.preserveInputs);
 
-  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div")] =
-      [this](const torch::jit::Node *node) { return loadDiv(node); };
-  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div_")] =
-      [this](const torch::jit::Node *node) { return loadDiv(node); };
+  for (auto &kv : valueMapReplacements) {
+    std::swap(valueMapReplacements[kv.first], valueMap_[kv.first]);
```

Comment:
```suggestion
    std::swap(kv.second, valueMap_[kv.first]);
```

Reply:
Also, why do you need to do this before *and* after loading the node? Shouldn't you do it only after loading the node?

Reply:
Technically we only know that an input should be constant-ified for a given node so to avoid contant-ifying other node's inputs I swap back after loading that node is done. In practice there isn't a difference probably because for example conv weights shouldn't get reused as inputs, it doesn't make sense but it's just to be correct.

Reply:
I think it would be useful to add a comment about this, then.